src/main/scala/translating/IRToBoogie.scala

```diff
@@ -475,27 +475,57 @@ class IRToBoogie(var program: Program, var spec: Specification) {
     def coalesced(): List[BExpr] = {
       val sections = memory.flatMap { s =>
         // Phrase the memory condition in terms of 64-bit operations, as long as the memory
-        // region is a multiple of such operations and appropriately aligned
-        if (s.address % 8 == 0 && s.bytes.size % 8 == 0) {
-          for (b <- s.bytes.indices by 8) yield {
-            // Combine the byte constants into a 64-bit value
-            val sum: BigInt =
-              (0 until 8).foldLeft(BigInt(0))((x, y) => x + (s.bytes(b + y).value * (BigInt(2).pow(y * 8))))
+        // section is a multiple of 64-bits and appropriately aligned
+
+        val aligned = s.address % 8
+        val alignedStart = s.address + aligned
+        val alignedSizeMultiple = (s.bytes.size - aligned) % 8
+        val alignedEnd = s.bytes.size - alignedSizeMultiple
+
+        // Section that is safe to convert to 64-bit values
+        val section = for (b <- aligned until alignedEnd by 8) yield {
+          // Combine the byte constants into a 64-bit value
+          val sum: BigInt =
```

Comment:
This is a concat not really a sum although it uses sums

Reply:
This name was just taken from the original implementation